client-v2/src/services/capiQuery.ts

```diff
@@ -31,9 +31,56 @@ interface CAPIOptions {
 interface CAPITagQueryReponse {
   response: {
     results: Tag[];
+    status: CAPIStatus;
+    message?: string;
   };
 }
 
+const getErrorMessageFromResponse = (response: Response) =>
+  `Error making a request to CAPI: the server returned ${response.status}, ${
+    response.statusText
+  }`;
+
+/**
+ * Fetch a CAPI response.
+ *
+ * @throws If the response fails for any reason.
+ */
+const fetchCAPIResponse = async <
+  TCAPIResponse extends CAPISearchQueryReponse | CAPITagQueryReponse
+>(
+  fetch: Fetch = window.fetch,
+  request: string
+) => {
+  let response: Response;
+  try {
+    response = await fetch(request);
+  } catch (e) {
+    if (e.status && e.statusText) {
```

Comment:
This looks like it expects a thrown `Response` object - is that a thing?

Reply:
This is a really interesting question, because that's what I _get_ -- but now that you'd said this, it's not what the spec seems to say (https://fetch.spec.whatwg.org/#fetch-method). All of the rejections seem to be passed descendants of an Error type.

I'm ... a bit confused. I'd be surprised if whatwg-fetch was behaving in a non-spec way, so it's probably something I'm not doing right in the tests -- I'll take a look.

Reply:
-- turns out we're using pandaFetch here, which is _not_ spec equivalent to window.fetch 🙃